internal/controllers/core/cmd/controller.go

```diff
@@ -311,7 +339,10 @@ func (c *Controller) runInternal(ctx context.Context,
 
 	stillHasSameProcNum := proc.stillHasSameProcNum()
 	c.updateStatus(name, func(status *CmdStatus) {
-		*status = CmdStatus{Waiting: &CmdStateWaiting{}}
+		*status = CmdStatus{
```

Comment:
seems like we should change this to use

status.Waiting = &CmdStateWaiting{}
status.Running = nil
status.Terminated = nil

like we do elsewhere in this file. i.e., only modify the fields we want to modify.